Doc/howto/sockets.rst

```diff
@@ -252,20 +252,23 @@ Binary Data
 -----------
 
 It is perfectly possible to send binary data over a socket. The major problem is
-that not all machines use the same formats for binary data. For example, a
-Motorola chip will represent a 16 bit integer with the value 1 as the two hex
-bytes 00 01. Intel and DEC, however, are byte-reversed - that same 1 is 01 00.
+that not all machines use the same formats for binary data. For example,
+network byte order is big-endian, with the most significant byte first,
```

Comment:
It is not clear why the network byte order is big-endian at first place.

What big-endian platforms are still in use?

Reply:
> It is not clear why the network byte order is big-endian at first place.

I'm not sure its within the scope of this how-to doc to describe the origins and history of network byte order, but to summarize, most CPU arches at the time except for Intel used it (which was a primary driver of the later rise of little-ended, out of compatibility rather than technical superiority), since it was more "natural", and in particular for network operations transmitting all the bits "in order" was necessary/helpful for some early forms of transmission, encoding and error correction schemes.

> What big-endian platforms are still in use?

As mentioned elsewhere in these changes, IBM z/Architecture, IBM Power, (Tier 2/3 Python platforms) SPARC, (Solaris) many microcontrollers and various other smaller ones, and RISC-V, ARM, MIPS (Longsoon, Sunway, etc) and others are bi-endian (though mostly little-endian in actual CPU designs). But as mentioned in the section, little-endian predominates in the consumer CPU space.

Reply:
I know, but it is not clear for the reader. The old text mentions a concrete big-endian processor. Maybe just replace it with a more modern example? Or at least add "for historical reasons" when mention the network byte order. Or better leave both, the user should know that the native order can be different from little-endian.

Reply:
For better or for worse, there is no real modern example, at least that more than a infinitesimal fraction of modern Python installations run on. The only architecture Python supports even at Tier 3 level that is not little-endian is IBM s390x mainframes running Linux, with deployments on the order of perhaps up to a few thousand total installs, compared with many millions to perhaps nearly on the order of a billion little-endian-CPU machines with some form of Python.

To note, it could equally be argued that the current prevalence of little-endian on the CPU side is also for historical reasons, not technical merit and despite big-endian being much more natural and logical, due to the historical dominance of Intel x86 CPUs, the resulting pressure to ease compatibility with software developed for them and the subsequent network effect that essentially required all new CPUs to be little-endian to have a chance at success in the market.

In any case, this is (supposed to be, at least; we probably need to refactor it further) a "how-to" guide, which per the [Diataxis framework we've adopted for the Python docs](https://diataxis.fr/), is concerned (naturally) with the "how" to practically accomplish a set of tasks, not the "why". The historical details behind why network byte order is big-endian are not necessary, or directly helpful to users looking for how to practically accomplish socket-related tasks in Python.  Users curious about such a topic are free to consult a more appropriate resource, such as Wikipedia. However, I've linked `network byte order` to the latter to make that easier to do.